BFF.py

```diff
@@ -267,6 +269,22 @@ def get_narrative_users(WorkspaceIdentityList, narrative_list, auth_token):
         narrative['users'] = response_json['result'][0]['perms'][index]
 
 
+@app.route('/fetchUserProfiles', methods=['POST'])
+def get_userProfiles():
+    body = request.get_json()
+    userIDs = body.get("users", [])
+    userProfile_payload = {
+        'id': 0,
+        'method': 'UserProfile.get_user_profile',
+        'version': '1.1',
+        'params': [userIDs]
+    }
+    user_profile_rpc_url = conf['KBASE_ENDPOINT'] + '/user_profile/rpc'
+    response = requests.post(
+        user_profile_rpc_url, data=json.dumps(userProfile_payload)
+    )
+    return json.dumps(response.json()), 200
```

Comment:
The response from the upstream user profile service will be jsonrpc, so it needs to be be unwrapped, as the singular get_user_profile does .I agree that it should not replicate the singular's behavior by creating a blank user profile rather than returning null, but just that the result should be unwrapped.

And I realize the codebase doesn't do this now, but it should validate the jsonrpc result as well, at least the jsonrpc structure (using a unique id in the request, verifying it is the same in the response, ensuring the jsonrpc version matches (we do have jsonrpc 2.0 at kbase too.) 

Errors should be caught too. The upstream jsonrpc error should be detected (it will be a 500 status and have the jsonrpc error structure), as well as a plain invalid response (some upstream errors like a 502 will result in both a non-200 status and also non-json response), and recast as meaningful bff errors.

As this is not a jsonrpc service, there is no uniform way to return errors. The registered error handlers return both json and text.

Reply:
I'm not familiar with the benefits of jsonrpc 2.0 over 1.1, what would they be in this case?

I think in this case I will pick the most appropriate HTTP status code and use that for indicating errors, returning any usable user profile data in the array, or else a list of nulls of the same length as the array in the request.